pysmt/solvers/msat.py

```diff
@@ -116,8 +160,63 @@ def solve(self, assumptions=None):
 
         assert res in [mathsat.MSAT_UNKNOWN,mathsat.MSAT_SAT,mathsat.MSAT_UNSAT]
         if res == mathsat.MSAT_UNKNOWN:
-            raise SolverReturnedUnknownResultError()
-        return res == mathsat.MSAT_SAT
+            raise SolverReturnedUnknownResultError
+
+        return (res == mathsat.MSAT_SAT)
+
+
+    def _check_unsat_core_config(self):
+        if self.options.unsat_cores_mode is None:
+            raise SolverNotConfiguredForUnsatCoresError
+
+        if self.last_result != False:
+            raise SolverStatusError("The last call to solve() was not" \
+                                    " unsatisfiable")
+
+        if self.last_command != "solve":
+            raise SolverStatusError("The solver status has been modified by a" \
+                                    " '%s' command after the last call to" \
+                                    " solve()" % self.last_command)
+
+
+    def get_unsat_core(self):
+        """After a call to solve() yielding UNSAT, returns the unsat core as a
+        set of formulae"""
+        if self.options.unsat_cores_mode == "all":
+            self._check_unsat_core_config()
```

Comment:
This is done on purpose, to avoid double checking. In fact, the same function ```self._check_unsat_core_config()``` is called from within ```get_named_unsat_core()```